Dynatrace/dynatrace-operator
# Support all custom pull secret auth methods

# Description

Currently only custom pull secrets with a username and a password are supported. 
This PR changes logic to use the docker client library to handle pull secrets in order to support all `dockerconfigjson` formats.

## How can this be tested?

- Create a custom pull secret for a private registry that does not use username and password (e.g. gcr)
- Set pull secret and image in the dynakube 
- The operator should be able to pull the image (visible in the logs)


## Checklist
